crates/transaction-pool/src/batcher.rs

```diff
@@ -0,0 +1,273 @@
+//! Transaction batching for `Pool` insertion for high-throughput scenarios
+//!
+//! This module provides transaction batching logic to reduce lock contention when processing
+//! many concurrent transaction pool insertions.
+
+use crate::{AddedTransactionOutcome, PoolTransaction, TransactionOrigin, TransactionPool};
+use alloy_primitives::B256;
+use pin_project::pin_project;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::sync::{mpsc, oneshot};
+
+/// Configuration for tx pool batch insertion
+#[derive(Debug, Clone)]
+pub struct BatchTxConfig {
+    /// Channel buffer size for incoming batch tx requests
+    pub channel_buffer_size: usize,
+    /// Maximum number of transactions to batch insert at a time
+    pub max_batch_size: usize,
```

Comment:
I think we only really need the batch size setting here?

the buffer size seems not redundant, because I think we can use an unbounded channel here, because the tx is already available on the request level and restricting the channel size doesnt really restrict anything if we drain + spawn, so we can use an unbounded channel instead